### ISSUE TITLE
Invasive test failed in version 1.5.1

### ISSUE THREAD

**Issue by simon28li** [libgit2/libgit2#6502]
Hi as discussed earlier [#6355](https://github.com/libgit2/libgit2/issues/6355)， invasive failed in the test of a new version of issue,1.5.1
### Reproduction steps
ctest  -V
### Expected behavior
All test pass
### Actual behavior
The following tests FAILED:
          2 - invasive (Failed)
### Version of libgit2 (release number or SHA1)
1.5.1
### Operating system(s) tested
centos8
Linux simon28li 4.18.0-383.el8.aarch64 #1 SMP Wed Apr 20 15:39:57 UTC 2022 aarch64 aarch64 aarch64 GNU/Linux
<details>
<summary>error msg</summary>
<pre>
<code>
1: worktree::config..
1: worktree::merge...
1: worktree::open........
1: worktree::reflog....
1: worktree::refs............
1: worktree::repository....
1: worktree::submodule....
1: worktree::worktree...................................
1:
1:   1) Failure:
1: remote::httpproxy::config_overrides [/home/li/stage/spack-stage-libgit2-1.5.1-zl5f3jjol3vgcyallcmt2rjmfliqoebc/spack-src/tests/libgit2/remote/httpproxy.c:74]
1:   Pointer mismatch: proxy != expected
1:   0x2e974f40 != (nil)
1:
1:   2) Failure:
1: remote::httpproxy::config_overrides_detached_remote [/home/li/stage/spack-stage-libgit2-1.5.1-zl5f3jjol3vgcyallcmt2rjmfliqoebc/spack-src/tests/libgit2/remote/httpproxy.c:127]
1:   Pointer mismatch: proxy != expected
1:   0x2e868d10 != (nil)
1:
1:   3) Failure:
1: remote::httpproxy::env [/home/li/stage/spack-stage-libgit2-1.5.1-zl5f3jjol3vgcyallcmt2rjmfliqoebc/spack-src/tests/libgit2/remote/httpproxy.c:54]
1:   Pointer mismatch: proxy != expected
1:   0x2ea61900 != (nil)
1:
1/9 Test #1: offline ..........................***Failed  188.58 sec
</code>
</pre>
</details>

**Comment by ethomson** [comment]
What does your ENV look like? Do you have an HTTP_PROXY set?

**Comment by simon28li** [comment]
> 你的环境是什么样的？你有设置 HTTP_PROXY 吗？

There are agents set up in the environment.
```
declare -x _sp_initializing
declare -x http_proxy="http://172.19.123.160:3128"
declare -x https_proxy="http://172.19.123.160:3128"
declare -x no_proxy="localhost,127.0.0.1,172.19.*"
```

**Comment by ethomson** [comment]
Thanks. I think I fixed this in #6498 which will arrive in v1.6.0 this weekend.

**Comment by simon28li** [comment]
> Thanks. I think I fixed this in #6498 which will arrive in v1.6.0 this weekend.

I tested version 1.6.1 and invasive has been executed successfully.
In addition, I also found that util execution failed when testing version 1.5.1\ 1.6.1. I executed under the root user, but I was successful when I switched to another non-root user. Does this testcase have to be executed under non-root users? Is it possible to be compatible with root users?

root:
```
5: utf8.
5: vector..........
5: wildmatch........
5: zstream....
5:
5:   1) Failure:
5: path::validate_current_user_ownership [/home/stage/root/spack-stage-libgit2-1.5.1-zl5f3jjol3vgcyallcmt2rjmfliqoebc/spack-src/tests/util/path.c:760]
5:   is_cur != 0
5:   1 != 0
5:
1/1 Test #5: util .............................***Failed    2.38 sec

0% tests passed, 1 tests failed out of 1

Total Test time (real) =   2.39 sec

The following tests FAILED:
          5 - util (Failed)
```
non-root:
```
12: vector..........
12: wildmatch........
12: zstream....
12:
1/1 Test #12: util .............................   Passed    2.43 sec

The following tests passed:
        util

100% tests passed, 0 tests failed out of 1

Total Test time (real) =   2.43 sec
```

**Comment by ethomson** [comment]
I think that this is fixed in #6513 and will let `root` run the ownership tests.

**Comment by ethomson** [comment]
#6513 was released in v1.6.3. Closing this - please let me know if you still see problems in your environment.
